pg_variables.c

```diff
@@ -1596,6 +1620,7 @@ createVariableInternal(Package *package, text *name, Oid typid, bool is_record,
 		variable->package = package;
 		variable->is_record = is_record;
 		variable->is_transactional = is_transactional;
+		variable->is_deleted = false;
```

Comment:
Why do we need this `is_deleted` flag? Is it required to make this deletion transactional?

Reply:
Yep, I think so. The trick here is that deletion is pretty much ok, but we get an error on insert a new variable with the same name since it still present in "memory contexts". It's lot easier to solve this particular problem with kinda lazy deletion than try clear contexts. So, we use this flag upon variable insertion to reinit record in case it was mark as deleted.